Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py

```diff
@@ -75,20 +75,44 @@ def __init__(self, tenant_id: str = '',
         self.auth_type = SELF_DEPLOYED_AUTH_TYPE if self_deployed else OPROXY_AUTH_TYPE
         self.verify = verify
 
-    def http_request(self, *args, **kwargs):
+    def http_request(self, headers=None, *args, **kwargs):
         """
         Overrides Base client request function, retrieves and adds to headers access token before sending the request.
 
         Returns:
             requests.Response: The http response
         """
         token = self.get_access_token()
-        headers = {
+        default_headers = {
             'Authorization': f'Bearer {token}',
             'Content-Type': 'application/json',
             'Accept': 'application/json'
         }
-        return super()._http_request(*args, headers=headers, **kwargs)  # type: ignore[misc]
+        if headers:
+            default_headers.update(headers)
+
+        response = super()._http_request(   # type: ignore[misc]
+            resp_type="response", headers=default_headers, *args, **kwargs)
+
+        # 206 indicates Partial Content, reason will be in the warning header.
+        # In that case, logs with the warning header will be written.
+        if response.status_code == 206:
+            demisto.debug(str(response.headers))
```

Comment:
could headers contain personal data? better print only the relevant keys

Reply:
No, in this case, it would contain fields that relate to the partial range download, such as `Content-Range` or `Content-Type`.
e.g:

```
HTTP/1.1 206 Partial Content
Content-Range: bytes 0-1023/2048
Content-Type: application/octet-stream

<first 1024 bytes of file>
```